agent/api/statechange.go

```diff
@@ -377,3 +377,72 @@ func (ts TaskStateChange) GetEventType() statechange.EventType {
 func (AttachmentStateChange) GetEventType() statechange.EventType {
 	return statechange.AttachmentEvent
 }
+
+// Implementation of the ContainerStateChange ContainerMetadataGetter Interface.
+type containerMetadataGetter struct {
```

Comment:
I'm thinking it might be better to put these metadata getters (i.e., `containerMetadataGetter`, `taskMetadataGetter`) in new file `agent/api/metadata_getter.go` and their tests in new file `agent/api/metadata_getter_test.go` for organizational purposes (i.e., they will still belong to the `api` package defined in agent module, but categorized into these files separate from `statechange.go` and `statechange_test.go` and thus can be found easier). What do you think?

Reply:
My understanding is that statechange.go would be moved entirely to `ecs-agent`, and the remaining methods were helpers for those structs. These metadata setters are a part of the statechange struct.

But I am happy to move them, since I will already push a new rev I can do this as well.